election-server/src/bin/http.rs

```diff
@@ -296,10 +331,22 @@ async fn setup_http(
     config: &AppConfig,
     prometheus_layer: PrometheusLayer,
 ) -> Result<tokio::task::JoinHandle<Result<(), anyhow::Error>>, anyhow::Error> {
+    let reader = csv::Reader::from_path(&config.eligible_voters_file)
+        .context("Failed to read eligible voters file")?;
+    let initial_weights = reader
+        .into_deserialize::<WeightRow>()
+        .try_fold(HashMap::new(), |mut initial_weights, row| {
+            let WeightRow { account, amount } = row?;
+            initial_weights.insert(account, amount);
+            anyhow::Ok(initial_weights)
+        })
+        .context("Failed to deserialize eligible voters file")?;
```

Comment:
```suggestion
    let initial_weights = {
        let mut initial_weights = HashMap::new();
        for row in reader.into_deserialize::<WeightRow>() {
            let WeightRow { account, amount } = row.context("Unable to parse initial weights row.")?;
            initial_weights.insert(account, amount);
        }
        initial_weights
    };
```

Or you could do more spelunking with your combinators

```rust
reader.into_deserialize::<WeightRow>().map(|r| r.map(|WeightRow{account, amount}| (account, amount))).collect::<Result<HashMap<_,_>,_>()?
```

I would really prefer a for loop solution. It's most easily maintainable. For example suppose you want to add a check that there are no duplicate rows (rows with account repeated). It's easy to modify.

Reply:
"Spelunking" 😆